### ISSUE TITLE
random numbers instead of token

### ISSUE THREAD

**Issue by rtiwarihr** [anx-ckreuzberger/django-rest-passwordreset#14]
Hello sir,
I am very impressed with this module and decided to use it. I would like to override the functionality where i can change the key 
def generate_key():
        """ generates a pseudo random code using os.urandom and binascii.hexlify """
        return binascii.hexlify(os.urandom(32)).decode()

to randint. 
Please let me know if there is a way to override it?

**Comment by anx-ckreuzberger** [comment]
You could, in theory, [monkey-patch](https://stackoverflow.com/questions/5626193/what-is-monkey-patching) the ``ResetPasswordToken`` model like this:

```python
import random
from django_rest_passwordreset.models import ResetPasswordToken

def my_generate_key():
    return random.randint(1000, 9999)

ResetPasswordToken.generate_key = my_generate_key
```

Please do not use this exact implementation of ``my_generate_key`` provided by me, as I'm not very happy with the security features that it provides.

Also, depending on the length of your generated key, you are going to be subject to brute force attacks that you need to handle somehow (e.g., by adding [Throttling](https://www.django-rest-framework.org/api-guide/throttling/)).

**Comment by iagocanalejas** [comment]
Would you accept a PR adding a configuration to django settings with an inportable string?

Thinking of a class with a method `get` to generate a new key and `parse` to convert the CharField to whatever you want (PE: int).

**Comment by anx-ckreuzberger** [comment]
That sounds like a good idea, we could rework the package to have a class ``BaseTokenGenerator`` with some basic implementations ``StringTokenGenerator`` and ``NumberTokenGenerator`` as well as configuration options for those classes (e.g., length).

Though I believe this is something that someone else must have already come up with, so I would like to see an existing reference implementation before we re-invent the wheel here.

**Comment by iagocanalejas** [comment]
You can check how to manage importable string for strategy pattern on django-rest-framework repository, [pagination](https://github.com/encode/django-rest-framework/blob/master/rest_framework/pagination.py) is a good example. 

DRF has a lot of configurable parametters but for this puspose the logic can be simplified.

**Comment by anx-ckreuzberger** [comment]
So we could come up with something like this:

```python
DJANGO_REST_PASSWORDRESET_TOKEN_CONFIG = {
    'CLASS': 'django_rest_passwordreset.tokens.StringTokenGenerator,
    'OPTIONS': {
        'min_length': 10,
        'max_length': 50
    }
}
```
and
```python
DJANGO_REST_PASSWORDRESET_TOKEN_CONFIG = {
    'CLASS': 'django_rest_passwordreset.tokens.NumberTokenGenerator,
    'OPTIONS': {
        'min_number': 1000,
        'max_number': 5000
    }
}
```

All keys in ``OPTIONS`` could be added when initializing the class as kwargs (using ``**OPTIONS``).

This would allow users to overwrite the class, and we would also not be limited by the configuration options for each class. The implementation of the classes would look like this:

```python
class StringTokenGenerator(BaseTokenGenerator):
    def __init__(min_length=15, max_length=20, *args, **kwargs):
        self.min_length = min_length
        self.max_length = max_length

    def generate_token():
        """ generates a pseudo random code using os.urandom and binascii.hexlify """
        return binascii.hexlify(os.urandom(rando‌​m.randint(self.min_length, self.max_length))).decode()
```

Is everyone happy with that solution?

**Comment by ChristianKreuzberger** [comment]
I've created a PR for this: #20

**Comment by anx-ckreuzberger** [comment]
Changes for this have just released in [Version 1.0.0a1](https://pypi.org/project/django-rest-passwordreset/1.0.0a1/).
